src/System.Net.Requests/src/System/Net/HttpDateParse.cs

```diff
@@ -0,0 +1,465 @@
+                second = rgdwDateParseResults[DATE_ANSI_INDEX_SECS];
+                if (iLastLettered != DATE_ANSI_INDEX_YEAR) {
+                    year   = rgdwDateParseResults[DATE_ANSI_INDEX_YEAR];
+                }
+                else {
+                    // This is a fix to get around toString/toGMTstring (where the timezone is
+                    // appended at the end. (See above)
+                    year   = rgdwDateParseResults[DATE_INDEX_TZ];
+                }
+            }
+            else {
+                day    = rgdwDateParseResults[DATE_1123_INDEX_DAY];
+                month  = rgdwDateParseResults[DATE_1123_INDEX_MONTH];
+                year   = rgdwDateParseResults[DATE_1123_INDEX_YEAR];
+                hour   = rgdwDateParseResults[DATE_1123_INDEX_HRS];
+                minute = rgdwDateParseResults[DATE_1123_INDEX_MINS];
+                second = rgdwDateParseResults[DATE_1123_INDEX_SECS];
+            }
+
+            //
+            // Normalize the year, 90 == 1990, handle the year 2000, 02 == 2002
+            //  This is Year 2000 handling folks!!!  We get this wrong and
+            //  we all look bad.
+            //
+
+            if (year < 100) {
+                year += ((year < 80) ? 2000 : 1900);
+            }
+
+            //
+            // if we got misformed time, then plug in the current time
+            // !lpszHrs || !lpszMins || !lpszSec
+            //
+
+            if ((i < 4)
+                || (day > 31)
+                || (hour > 23)
+                || (minute > 59)
+                || (second > 59)) {
+                fRet = false;
+                goto quit;
+            }
+
+            //
+            // Now do the DateTime conversion
+            //
+
+            dtOut = new DateTime (year, month, day, hour, minute, second, millisecond);
+
+            //
+            // we want the system time to be accurate. This is _suhlow_
+            // The time passed in is in the local time zone; we have to convert this into GMT.
+            //
+
+            if (iLastLettered==DATE_ANSI_INDEX_YEAR) {
+                // this should be an unusual case.
+                dtOut = dtOut.ToUniversalTime();
+            }
+
+            //
+            // If we have an Offset to another Time Zone
+            //   then convert to appropriate GMT time
+            //
+
+            if ((i > DATE_INDEX_TZ &&
+                 rgdwDateParseResults[DATE_INDEX_TZ] != DATE_TOKEN_GMT)) {
+
+                //
+                // if we received +/-nnnn as offset (hhmm), modify the output FILETIME
+                //
+
+                double offset;
+
+                offset = (double) rgdwDateParseResults[DATE_INDEX_TZ];
+                dtOut.AddHours(offset);
+            }
+
+            // In the end, we leave it all in LocalTime
+
+            dtOut = dtOut.ToLocalTime();
+
+            quit:
+
+            return fRet;
+        }
+    }
+
+} // namespace System.Net
```

Comment:
Why are we doing this parsing?  We should be able to just use .NET DateTime.Parse() routines.